packages/next/src/client/components/navigation.ts

```diff
@@ -211,7 +211,10 @@ function useSelectedLayoutSegments(
   parallelRouteKey: string = 'children'
 ): string[] {
   clientHookInServerComponentError('useSelectedLayoutSegments')
-  const { tree } = useContext(LayoutRouterContext)
+  const tree = useContext(LayoutRouterContext)?.tree
+  // @ts-expect-error This only happens in `pages`. Type is overwritten in navigation.d.ts
```

Comment:
```suggestion
  // This only happens in `pages`. Type is overwritten in navigation.d.ts
```

Reply:
See https://github.com/vercel/next.js/pull/62584#discussion_r1507239370

Reply:
we can make `LayoutRouterContext` 's value could also return null, like what we did for `SearchParamsContext` and `AppRouterContext`. 

```ts
export const LayoutRouterContext = React.createContext<{
  childNodes: CacheNode['parallelRoutes']
  tree: FlightRouterState
  url: string
} | null>(null as any)
```

then we can return null when context is null


```ts
function useSelectedLayoutSegments(
  parallelRouteKey: string = 'children'
): string[] {
  clientHookInServerComponentError('useSelectedLayoutSegments')
  const layoutRouterContext = useContext(LayoutRouterContext)
  
  if (!layoutRouterContext) {
    // for pages router
    return null as unknown as string[]
  }

  return getSelectedLayoutSegmentPath(layoutRouterContext.tree, parallelRouteKey)
}
```